### PR TITLE
Make mp.Queue.get  async for response queue

### PR DESCRIPTION
**Pull request by aniketmaurya**
<details>
  <summary><b>Before submitting</b></summary>

- [ ] Was this discussed/agreed via a Github issue? (no need for typos and docs improvements)
- [ ] Did you read the [contributor guideline](https://github.com/Lightning-AI/pytorch-lightning/blob/main/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to update the docs?
- [ ] Did you write any new necessary tests?

</details>

- [x] Rerun performance 
- [x] Update tests

Improves throughput.

Fixes # (issue).

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in GitHub issues there's a high chance it will not be merged.

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Review by lantiga — APPROVED** [review]
Nice one. Let's rerun all benchmarks : )

File: .github/workflows/ci-testing.yml [hunk 1]
```diff
@@ -63,7 +63,9 @@ jobs:
           pip list
 
       - name: Tests
-        run: coverage run --source litserve -m pytest src tests -v
+        run: |
+          coverage run --source litserve -m pytest src tests -v
+          echo "Test completed!"
```
Comment (Borda) [inline .github/workflows/ci-testing.yml:68]:
```suggestion
          python -m pytest --cov=litserve src/ tests/ -v
```

File: .github/workflows/ci-testing.yml [hunk 2]
```diff
@@ -63,7 +63,8 @@ jobs:
           pip list
 
       - name: Tests
-        run: coverage run --source litserve -m pytest src tests -v
+        run: |
+          python -m pytest --cov=litserve src/ tests/ -v
```
Comment (Borda) [inline .github/workflows/ci-testing.yml:67]:
```suggestion
          python -m pytest --cov=litserve src/ tests/ -v -s
        timeout-minutes: 10
```

File: .github/workflows/ci-testing.yml [hunk 3]
```diff
@@ -63,7 +63,9 @@ jobs:
           pip list
 
       - name: Tests
-        run: coverage run --source litserve -m pytest src tests -v
+        run: |
+          python -m pytest --cov=litserve src/ tests/ -v -s
```
Comment (Borda) [inline .github/workflows/ci-testing.yml:67]:
```suggestion
          python -m pytest --cov=litserve src/ tests/ -v --capture=no
```

Comment (aniketmaurya) [reply .github/workflows/ci-testing.yml:67]: @Borda found out the reason CI seem to be stuck - https://github.com/pytest-dev/pytest-asyncio/issues/179

Comment (Borda) [reply .github/workflows/ci-testing.yml:67]: also could be related to https://github.com/miguelgrinberg/python-socketio/issues/263

Comment (Borda) [reply .github/workflows/ci-testing.yml:67]: btw, I can reproduce the hanging also locally

Comment (aniketmaurya) [reply .github/workflows/ci-testing.yml:67]: thanks for checking @Borda! yeah, I can also reproduce locally now. [This](https://github.com/Lightning-AI/LitServe/blob/f6833237215cc62d8dafc0d2280fb2306879d0f9/tests/test_lit_server.py#L402-L405) particular block of test is making CI fail.

Comment (aniketmaurya) [reply .github/workflows/ci-testing.yml:67]: https://github.com/Lightning-AI/LitServe/blob/f6833237215cc62d8dafc0d2280fb2306879d0f9/tests/test_lit_server.py#L402-L405

**Review by Borda — APPROVED** [review]

**Pull request merged**